posthog/api/dashboard.py

```diff
@@ -219,8 +220,18 @@ def get_queryset(self) -> QuerySet:
         if self.action != "update":
             # Soft-deleted dashboards can be brought back with a PATCH request
             queryset = queryset.filter(deleted=False)
-        queryset = queryset.select_related("team__organization").prefetch_related(
-            Prefetch("items", queryset=Insight.objects.filter(deleted=False).order_by("order"))
+
+        deferred_fields = set.union(
+            get_deferred_field_set_for_model(
+                Organization, fields_not_deferred={"available_features"}, field_prefix="team__organization__"
+            ),
+            get_deferred_field_set_for_model(Team, fields_not_deferred={"organization", "name"}, field_prefix="team__"),
+        )
+
+        queryset = (
+            queryset.select_related("team__organization", "created_by")
+            .defer(*deferred_fields)
```

Comment:
There is also an “only” option as well, although I haven’t used it, would that be appropriate?

Would also consider just using values/values_list so it’s super explicit and no field is accidentally accessed, but not sure how that would play with the calling functions

Reply:
They marked it as a "won't fix" here: https://code.djangoproject.com/ticket/29072 😔

Reply:
How annoying. What does dashboard serializer need all fields for?

Reply:
non-blocking

Also, would there be an issue with listing out the fields? I have a preference for being explicit and dumb if possible. Might not be relevant at this level but I can imagine the related REST API knows exactly the fields it should be returning.

Reply:
> What does dashboard serializer need all fields for?

It might not be all fields, but it's nearly all to populate the data sent down to the client.

> Also, would there be an issue with listing out the fields? I have a preference for being explicit and dumb if possible. Might not be relevant at this level but I can imagine the related REST API knows exactly the fields it should be returning.

This is definitely a fair question

The issue that I've seen with being too explicit with the fields you're requesting in a queryset is that when you add a new field to the model/serializer and not to that queryset (which inevitably happens unless we add strong protections), Django will make the standard list query from the database (excluding the new field), and then it will make an additional query for each individual row to populate the missing field in the serializer. I've done this a time or two at old workplaces and massively slowed down production... 😬. 

We could lean on tests like the on in this PR to catch these issues, but we would want full coverage of our endpoints, and if someone adds a new serializer that isn't tested in this way, it's really easy to hit this issue.

With that said, I see the trade-offs here, and the benefits of being explicit. But I think at our level of scale on the application side of the product (e.g. dashboards, insights etc.), it feels like we'd be adding complexity to our codebase for a problem we aren't really hitting yet.